util/queries.js

```diff
@@ -13,50 +13,100 @@ export async function getResourceInfo(uri) {
         BIND (${sparqlEscapeUri(uri)} as ?resource)
         ?resource a ?type .
       }
+      FILTER NOT EXISTS {
+       ?resource <http://schema.org/publication> ?publicationStatus.
+      }
       FILTER(?g NOT IN (<http://redpencil.data.gift/id/deltas/producer/loket-submissions>))
     }
     LIMIT 1`);
 
   if (result.results.bindings.length) {
     return result.results.bindings;
   } else {
-    console.log(`Resource not found.`);
+    console.log(`Resource ${uri} not found or already published`);
     return null;
   }
 }
 
-export async function getSubmissionInfo(uri, pathToSubmission) {
-  let pathToForm = '';
-  if (pathToSubmission) {
-    pathToSubmission = pathToSubmission.replaceAll("?subject", sparqlEscapeUri(uri));
-    pathToForm = '?submission <http://www.w3.org/ns/prov#generated> ?form .';
-  } else {
-    pathToSubmission = '';
-    pathToForm = `${sparqlEscapeUri(uri)} <http://www.w3.org/ns/prov#generated> ?form .`;
+export async function getUnpublishedSubjectsFromSubmission(submission, type, pathToSubmission = ''){
+  //TODO:
+  // 1. This is extremely implict: the pathToSubmission expects the name `?subject` as root node, and `?submission` as submission
+  // 2. Re-think the black-listing of graphs.
+  const queryStr = `
+    SELECT DISTINCT ?subject WHERE {
+      BIND(${sparqlEscapeUri(submission)} as ?submission)
+
+      GRAPH ?g {
+        ?subject a ${sparqlEscapeUri(type)}.
+      }
+
+      ${pathToSubmission}
+
+     FILTER NOT EXISTS {
+       ?subject <http://schema.org/publication> ?publicationStatus.
+     }
+
+     FILTER(?g NOT IN (<http://redpencil.data.gift/id/deltas/producer/loket-submissions>))
+    }
+  `;
+
+  const result = await query(queryStr);
+  if (result.results.bindings.length) {
+    return result.results.bindings.map(r => r.subject.value);
   }
+  else {
+    console.log(`No unpublished subjects found for ${submission} and ${type}`);
+    return [];
+  }
+}
 
+export async function getSubmissionInfo(uri, pathToSubmission, type) {
+    if(type == 'http://www.w3.org/2004/02/skos/core#Concept'){
+      return null;
+    }
```

Comment:
Same as before, why aren't we publishing concepts ?

Reply:
The problem is basically, if this function is executed, it wil return all submissions linked to this concept. So we don't have the right one. So for the scheduling of the related resources, this might end up very ugly...

Reply:
Problem is, i don't see, expect hackish way, to export these.